src/scripts/main.js

```diff
@@ -1,3 +1,330 @@
+select.setAttribute('data-qa', 'office');
+select.required = true;
+optTokyo.textContent = 'Tokyo';
+optTokyo.setAttribute('value', 'Tokyo');
+optSingapore.textContent = 'Singapore';
+optSingapore.setAttribute('value', 'Singapore');
+optLondon.textContent = 'London';
+optLondon.setAttribute('value', 'London');
+optNY.textContent = 'New York';
+optNY.setAttribute('value', 'New York');
+optEdinburgh.textContent = 'Edinburgh';
+optEdinburgh.setAttribute('value', 'Edinburgh');
+optSF.textContent = 'San Francisco';
+optSF.setAttribute('value', 'San Francisco');
+select.append(optTokyo, optSingapore, optLondon, optNY, optEdinburgh, optSF);
+labelOffice.append(select);
+
+const labelAge = document.createElement('label');
+const inputAge = document.createElement('input');
+
+labelAge.textContent = 'Age:';
+inputAge.setAttribute('name', 'age');
+inputAge.setAttribute('type', 'number');
+inputAge.setAttribute('data-qa', 'age');
+inputAge.required = true;
+labelAge.append(inputAge);
+
+const labelSalary = document.createElement('label');
+const inputSalary = document.createElement('input');
+
+labelSalary.textContent = 'Salary:';
+inputSalary.setAttribute('name', 'salary');
+inputSalary.setAttribute('type', 'number');
+inputSalary.setAttribute('data-qa', 'salary');
+inputSalary.required = true;
+labelSalary.append(inputSalary);
+
+const button = document.createElement('button');
+
+button.setAttribute('type', 'submit');
+button.textContent = 'Save to table';
+
+form.className = 'new-employee-form';
+body.append(form);
+
+form.append(
+  labelName,
+  labelPosition,
+  labelOffice,
+  labelAge,
+  labelSalary,
+  button,
+);
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const person = {};
+  const inputs = form.querySelectorAll('input, select');
+
+  inputs.forEach((input) => {
+    person[input.name] = input.value;
+  });
+
+  const value = salaryToDollars(person.salary);
+
+  person['salary'] = value;
+
+  const pushNotification = (type) => {
+    const div = document.createElement('div');
+    const h2 = document.createElement('h2');
+    const p = document.createElement('p');
+
+    div.className = 'notification';
+    div.classList.add(type);
+    div.setAttribute('data-qa', 'notification');
+    div.style.visibility = 'visible';
+
+    if (type === 'error') {
+      h2.textContent = 'Error!!!';
+      h2.className = 'title';
+
+      p.textContent = `Your Name value has less than 4 letters or
+        your Age value is less than 18 or more than 90 or
+        some fild is missed`;
+    } else {
+      h2.textContent = 'Success!';
+      h2.className = 'title';
+
+      p.textContent = `Your data has been added to the table`;
+    }
+    div.append(h2, p);
+    body.append(div);
+
+    function hide(someDiv) {
+      someDiv.style.visibility = 'hidden';
+    }
+
+    setTimeout(() => hide(div), 5000);
+  };
+
+  if (
+    person.name.length < 4 ||
+    +person.age < 18 ||
+    +person.age > 90 ||
+    person.position === ''
```

Comment:
The validation logic checks if the position is an empty string but does not check for the office field. Ensure all required fields are validated before submission.